diff --git a/src/modules/config.js b/src/modules/config.js
index 3cd6888f..1b6d3416 100644
--- a/src/modules/config.js
+++ b/src/modules/config.js
@@ -50,7 +50,7 @@ export const defaultState = {
   autohideFloatingPostButton: false,
   pauseOnUnfocused: true,
   stopGifs: true,
-  replyVisibility: 'all',
+  replyVisibility: 'following',
   thirdColumnMode: 'notifications',
   notificationVisibility: {
     follows: true,
