diff --git a/lib/pleroma/web/activity_pub/activity_pub.ex b/lib/pleroma/web/activity_pub/activity_pub.ex
index 4c29dda35..29f5bb655 100644
--- a/lib/pleroma/web/activity_pub/activity_pub.ex
+++ b/lib/pleroma/web/activity_pub/activity_pub.ex
@@ -947,8 +947,7 @@ defp restrict_replies(query, %{
   end

   defp restrict_replies(query, %{
-         reply_filtering_user: %User{} = user,
-         reply_visibility: "following"
+         reply_filtering_user: %User{} = user
        }) do
     from(
       [activity, object] in query,
